### PR TITLE
Added ability to register the same handler function for different routes...

### PR DESCRIPTION
**Pull request by r0mant**
....

### CODE REVIEW

**Comment by r0mant** [comment]
@russjones @jeremyschlatter

File: app.go [hunk 1]
```diff
@@ -109,15 +109,21 @@ func (app *App) AddHandler(spec Spec) error {
 		return fmt.Errorf("the spec does not provide a handler function: %v", spec)
 	}
 
-	// register the handler in the router
-	route := app.router.HandleFunc(spec.Path, handler).Methods(spec.Methods...)
-	if len(spec.Headers) != 0 {
-		route.Headers(spec.Headers...)
+	// backward compatibility
+	if spec.Path != "" {
```
Comment (jeremyschlatter) [inline app.go]: How about

```go
if spec.Path != "" && spec.Paths == nil {
```

**Comment by jeremyschlatter** [comment]
left a suggestion, but lgtm either way

File: app.go (hunk 1 above)
Comment (r0mant) [inline app.go]: Sounds good. Done. Ideally these two should be mutually exclusive but it's good for now.

**Pull request merged**